eb_ssm/eb_ssm.py

```diff
@@ -76,25 +91,42 @@ def ssh(self):
             else:
                 io.log_error("The current Elastic Beanstalk application has no environments")
             sys.exit()
-        
+
         instances = get_instance_ids(self.environment_name)
-        if len(instances) == 1:
+        if len(instances) == 1 or self.non_interactive is True:
             instance = instances[0]
         else:
             io.echo()
             io.echo('Select an instance to ssh into')
             instance = utils.prompt_for_item_in_list(instances)
-        
+
         params = [
             "aws", "ssm", "start-session",
             "--document-name", "AWS-StartInteractiveCommand",
-            "--parameters", "command='bash -l'",
+            "--parameters", "command='" + self.command + "'",
             "--profile", self.profile,
             "--region", self.region,
             "--target", instance,
         ]
-        
-        os.system(" ".join(params))
+        cmd = " ".join(params)
+
+        # Filter the output of os.system(cmd to remove strange output from Session Manager in non_interactive mode.
+        # The output might begin with:
+        #
+        # Starting session with SessionId: [id]
+        #
+        # and end with:
+        # Exiting session with sessionId:  [id].
+        # @see https://github.com/aws/session-manager-plugin/pull/20
+        # @see https://github.com/aws/amazon-ssm-agent/issues/358
+        if self.non_interactive is True:
```

Comment:
I think it's better not to put temporary workarounds in eb-ssm given the underlying issues will (theoretically) be fixed soon.  I'd cut this from the PR.

Reply:
Ok, agreed, also it reads everything in the memory, for instance when dealing with DB dumps, it's nearly useless.